File: src/backend/BePro.Core/Entities/Placement.cs
using BePro.Core.Entities.Base;
using BePro.Core.Enums;

namespace BePro.Core.Entities;

public class Placement : BaseEntity
{
    public Guid CandidateId { get; set; }
    public Candidate Candidate { get; set; } = null!;

    public DateTime HireDate { get; set; }
    public DateTime? GuaranteeEndDate { get; set; }
    public bool? GuaranteeMet { get; set; }
    public DateTime? TerminationDate { get; set; }

    public PaymentStatus FreelancerPaymentStatus { get; set; } = PaymentStatus.Pending;
    public DateTime? FreelancerPaymentDate { get; set; }
}

File: packages/shared/src/types/placement.ts
export type PaymentStatus = "pending" | "paid" | "cancelled";

export interface IPlacementDto {
  id: string;
  candidateId: string;
  candidateFullName: string;
  hireDate: string;
  guaranteeEndDate?: string;
  guaranteeMet?: boolean;
  terminationDate?: string;
  freelancerPaymentStatus: PaymentStatus;
  freelancerPaymentDate?: string;
}

export interface ICreatePlacementRequest {
  candidateId: string;
  hireDate: string;
  guaranteeEndDate?: string;
}

export interface IUpdatePlacementRequest {
  guaranteeEndDate?: string;
  guaranteeMet?: boolean;
  terminationDate?: string;
  freelancerPaymentStatus?: PaymentStatus;
  freelancerPaymentDate?: string;
}
